src/core/auth/auth.controller.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Param,
+  Response,
+  InternalServerErrorException,
+  BadRequestException,
+  UnauthorizedException,
+  ForbiddenException,
+} from '@nestjs/common';
+import { AuthService } from './service/auth.service';
+import { LoginDto } from './dto/login.dto';
+import { RegisterDto } from './dto/register.dto';
+import { ApiOkResponse, ApiResponse, ApiTags } from '@nestjs/swagger';
+import * as express from 'express';
+import { ApiException } from '@nanogiants/nestjs-swagger-api-exception-decorator';
+import { SuccessfullyLogin, SuccessfullyRegister } from './dto/responses.dto';
+import { INTERNAL_SERVER_ERROR } from 'src/constants/messages/messagesConst';
+import { MessageDTO } from 'src/common/dto/response.dto';
+@ApiTags('auth')
+@Controller('auth')
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Post('register')
+  @ApiOkResponse({
+    description: 'Returns the created user',
+    type: SuccessfullyRegister,
+  })
+  @ApiException(() => BadRequestException, {
+    description: 'User already exists',
+  })
+  @ApiException(() => InternalServerErrorException, {
+    description: INTERNAL_SERVER_ERROR,
+  })
+  async register(@Body() registerDto: RegisterDto) {
+    const response = await this.authService.register(registerDto);
+    await this.authService.createEmailToken(registerDto.email);
+    const emailVerificationResponse = await this.sendEmailVerification(
+      registerDto.email,
+    );
+
+    return {
+      ...response,
+      emailVerification: emailVerificationResponse,
+    };
+  }
+
+  @Post('login')
+  @ApiOkResponse({
+    description: 'Returns the token',
+    type: SuccessfullyLogin,
+  })
+  @ApiException(() => UnauthorizedException, {
+    description: 'Email or password incorrect',
+  })
+  @ApiException(() => InternalServerErrorException, {
+    description: INTERNAL_SERVER_ERROR,
+  })
+  login(@Body() loginDto: LoginDto) {
+    return this.authService.login(loginDto);
+  }
+
+  @Get('email/resend-verification/:email')
+  @ApiOkResponse({
+    description: 'Returns the message',
+    type: MessageDTO,
+  })
+  @ApiException(() => ForbiddenException, {
+    description: 'user not found',
+  })
+  @ApiException(() => InternalServerErrorException, {
+    description: INTERNAL_SERVER_ERROR,
+  })
```

Comment:
Es un caso plausible y posible obtener un `INTERNAL_SERVER_ERROR` aqui?

Reply:
Mejor tenerlo y no necesitarlo que necesitarlo y no tenerlo

Reply:
Creo que es mas algo del repositorio que envio salcedo de guia,pero por lo menos en mi otro proyecto de io si me ha salido ese error

Reply:
Lo deje mas por error de acceso al servidor ya sea pq se cayo el backend en el vps, o sucedio algo grave en algun sitio y no te deje hacer solicitudes

Reply:
Entiendo, pero ese
```
  @ApiException(() => InternalServerErrorException, {
    description: INTERNAL_SERVER_ERROR,
  })
```
Se repite mucho en todo, por eso digo que es trivial, pudiesemos crear un decorador personalizado?, de modo que hagamos algo tipo
```suggestion
  @ApiExceptionInternalServerError()
```
Y ya ese decorador personalizado ponga lo demas el por dentro, para no repetir tanto codigo